src/mono/wasm/runtime/managed-exports.ts

```diff
@@ -38,6 +23,23 @@ export function init_managed_exports(): void {
     if (!runtimeHelpers.runtime_interop_exports_class)
         throw "Can't find " + runtimeHelpers.runtime_interop_namespace + "." + runtimeHelpers.runtime_interop_exports_classname + " class";
 
+    const install_sync_context = mono_method_resolve(
+        "[System.Runtime.InteropServices.JavaScript] System.Runtime.InteropServices.JavaScript.JSSynchronizationContext:InstallSynchronizationContext"
+    );
+    mono_assert(install_sync_context, "Can't find InstallSynchronizationContext method");
+    runtimeHelpers.javaScriptExports.install_synchronization_context = () => {
+        const sp = anyModule.stackSave();
+        try {
+            const args = alloc_stack_frame(2);
+            invoke_method_and_handle_exception(install_sync_context, args);
+        } finally {
+            anyModule.stackRestore(sp);
+        }
+    };
```

Comment:
So basically everything LGTM, except this part, which I don't understand. It seems like you're doing what the JSExport source generator is supposed to do, but by hand? but you're also using `mono_method_resolve` from the legacy API?  I'm not saying it's wrong - just that I don't know how to tell if it's right.  @pavelsavara can you take a look at this bit.

Reply:
`mono_method_resolve` makes sense to use instead of doing it by hand, even if we're not making it public anymore. I don't see any reason not to use it.

And yeah, I'm duplicating JSExport by hand in this case. Pavel said we do it for a few other things in the runtime and it's the most straightforward option here since we don't need any argument or return value marshaling.

Reply:
Makes sense, thanks!

Reply:
for the internal methods which need to work early during runtime startup, this is good option, to not use the generator etc.

Reply:
you should use simpler `get_method()` from `src\mono\wasm\runtime\managed-exports.ts` instead of `mono_method_resolve`.